src/Chapter4.hs

```diff
@@ -629,7 +662,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM a b = (a >>= (\x -> pure (x &&))) <*> b
```

Comment:
This fails when `a = Just False` and `b = Nothing`
shouldn't all computation involving `Nothing` returns `Nothing`?

Reply:
also tried with
```
andM a b = do
  m<-a
  n<-b
  return (m && n)
```
and simpler
```
fmap (&&) a <*>
```
still failing the case

Reply:
kinda fixed in the latest revision, not sure if the solution is correct tho

Reply:
If you don't use the `>>=` operator (like in the last implementation), then the `Monad` constraint is too strong, and it is enough to specify only `Applicative`. However, with `Applicative` you are not able to implement a lazier version of `andM`.

The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face:

Reply:
got it, thanks for the explanation (: